File: frontend/src/app/models/project.ts
import { GRAY_COLOR, hexToNumber } from '../shared/form/color-picker/color-picker.component';

export class Project {
	clientId: number;
	clientName: string;
	clientIsActive: boolean;
	color: number;
	daysBeforeStopEditTimeEntries: number;
	description: string;
	id: number;
	isActive: boolean;
	isCurrentUserOnProject: boolean;
	isNotificationEnabled: boolean;
	isPrivate: boolean;
	isTimeLockEnabled: boolean;
	lockPeriod: number;
	membersCount: number;
	name: string;
	notificationDay: number;
	tasksCount: number;

	constructor(data = null) {
		this.clientId = data && data.clientId;
		this.clientName = data && data.clientName;
		this.clientIsActive = data && data.clientIsActive;
		this.color = data && data.color || hexToNumber(GRAY_COLOR);
		this.daysBeforeStopEditTimeEntries = data && data.daysBeforeStopEditTimeEntries;
		this.description = data && data.description;
		this.id = data && data.id;
		this.isActive = data && data.isActive;
		this.isCurrentUserOnProject = data && data.isCurrentUserOnProject;
		this.isNotificationEnabled = data && data.isNotificationEnabled;
		this.isPrivate = data && data.isPrivate;
		this.isTimeLockEnabled = data && data.isTimeLockEnabled;
		this.lockPeriod = data && data.lockPeriod;
		this.notificationDay = data && data.notificationDay;
		this.membersCount = data && data.membersCount;
		this.name = data && data.name;
		this.tasksCount = data && data.tasksCount;
	}
}

File: backend/CoralTime.DAL/Models/Project.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CoralTime.Common.Constants;

namespace CoralTime.DAL.Models
{
    public class Project : LogChanges, IInitializeByName
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Name { get; set; }

        public int? ClientId { get; set; }

        [ForeignKey("ClientId")]
        public Client Client { get; set; }

        public int DaysBeforeStopEditTimeEntries { get; set; }

        public bool IsTimeLockEnabled { get; set; }

        public bool IsNotificationEnabled { get; set; }

        public List<TimeEntry> TimeEntries { get; set; }

        [MaxLength(500)]
        public string Description { get; set; }

        public bool IsPrivate { get; set; }

        public bool IsActive { get; set; }

        public bool IsActiveBeforeArchiving { get; set; }

        public int Color { get; set; }

        public List<MemberProjectRole> MemberProjectRoles { get; set; }

        public List<TaskType> TaskTypes { get; set; }

        public Constants.LockTimePeriod LockPeriod { get; set; }

        public int NotificationDay { get; set; }
    }
}